x-pack/plugins/lens/public/embeddable/embeddable.tsx

```diff
@@ -139,6 +157,52 @@ const getExpressionFromDocument = async (
   };
 };
 
+async function getViewUnderlyingDataArgs({
```

Comment:
I think this function can work without `async`. There's no need to return a `Promise`.

Reply:
If we remove the waiting, don't we run the risk of this function being called before the args are ready?

Reply:
Ah, my mistake. Wrong function. You're right 👍